src/main/java/net/snowflake/client/core/StmtUtil.java

```diff
@@ -85,6 +87,8 @@ static class StmtInput {
     int injectSocketTimeout; // seconds
     int injectClientPause; // seconds
 
+    int maxRetries = MAX_RETRIES;
```

Comment:
Do we have ability to pass max retries as a connection parameter? So if application needs 0 for any reason, they can set it. Is this consider as a BCR change as by default now we have fix retries?

Reply:
@sfc-gh-skumbham we should give user ability to go back to previous behavior. We can keep track of network timeout and max retires and terminate the retry effort whichever comes first. We shouldn't introduce any regressions and keep driver backward compatible.

Reply:
Even if you give the users the ability to go back to previous behavior it is still a behavior change and we should follow all the steps to treat it like one.